http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GceRuntimeMonitorInterp.scala

```diff
@@ -0,0 +1,690 @@
+  ctxConversion,
+  dbioToIO,
+  nowInstant,
+  userScriptStartupOutputUriMetadataKey
+}
+import org.broadinstitute.dsde.workbench.leonardo.model.LeoAuthProvider
+import org.broadinstitute.dsde.workbench.leonardo.monitor.GceRuntimeMonitor._
+import org.broadinstitute.dsde.workbench.leonardo.monitor.GcsMonitorState._
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage.{
+  CreateRuntimeMessage,
+  RuntimeTransitionMessage
+}
+import org.broadinstitute.dsde.workbench.leonardo.util._
+import org.broadinstitute.dsde.workbench.model.TraceId
+import org.broadinstitute.dsde.workbench.model.google.{GcsPath, GoogleProject}
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+import slick.dbio.DBIOAction
+
+import GceRuntimeMonitorInterp._
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+class GceRuntimeMonitorInterp[F[_]: Timer: Parallel](
+  config: GceMonitorConfig,
+  googleComputeService: GoogleComputeService[F],
+  publisherQueue: fs2.concurrent.InspectableQueue[F, LeoPubsubMessage],
+  authProvider: LeoAuthProvider[F],
+  googleStorageService: GoogleStorageService[F],
+  gceInterpreter: RuntimeAlgebra[F]
+)(implicit dbRef: DbReference[F],
+  runtimeToolToToolDao: RuntimeContainerServiceType => ToolDAO[F, RuntimeContainerServiceType],
+  F: Async[F],
+  logger: Logger[F],
+  ec: ExecutionContext,
+  openTelemetry: OpenTelemetryMetrics[F])
+    extends GceRuntimeMonitor[F] {
+  def process(runtimeId: Long)(implicit ev: ApplicativeAsk[F, TraceId]): Stream[F, Unit] =
+    for {
+      // building up a stream that will terminate when gce runtime is ready
+      traceId <- Stream.eval(ev.ask)
+      startMonitoring <- Stream.eval(nowInstant[F])
+      monitorContext = MonitorContext(startMonitoring, runtimeId, traceId)
+      // 20 seconds is from experience. No status transition took less than 20 seconds. Hence wait 20 seconds before start polling
+      _ <- Stream.sleep(config.initialDelay) ++ Stream.unfoldLoopEval[F, GcsMonitorState, Unit](Initial)(
+        s => Timer[F].sleep(config.pollingInterval) >> handler(monitorContext, s)
+      )
+    } yield ()
+
+  // Function used for transitions that we can get an Operation
+  def pollCheck(googleProject: GoogleProject,
+                runtimeAndRuntimeConfig: RuntimeAndRuntimeConfig,
+                operation: com.google.cloud.compute.v1.Operation,
+                action: RuntimeStatus)(implicit ev: ApplicativeAsk[F, TraceId]): F[Unit] =
+    for {
+      // building up a stream that will terminate when gce runtime is ready
+      traceId <- ev.ask
+      startMonitoring <- nowInstant
+      monitorContext = MonitorContext(startMonitoring, runtimeAndRuntimeConfig.runtime.id, traceId)
+      // 20 seconds is from experience. No status transition took less than 20 seconds. Hence wait 20 seconds before start polling
+      _ <- Timer[F].sleep(30 seconds)
```

Comment:
Move `30 seconds` to a config along with the comment (which probably needs updating).